src/sugar.jl

```diff
@@ -61,26 +61,54 @@ foldtree(op, init, x) = op(init, x)
 foldtree(op, init, ex::Expr) =
     op(foldl((acc, x) -> foldtree(op, acc, x), ex.args; init=init), ex)
 
-need_dynamic_lens(ex) =
-    foldtree(false, ex) do yes, x
-        yes || x === :end || x === :_
+@static if VERSION ≥ v"1.5.0-DEV.666"
```

Comment:
I think instead of duplicating functions, I think I would do something like
```julia
const HAS_BEGIN_INDEXING = VERSION ≥ v"1.5.0-DEV.666"

function need_dynamic_lens(ex)
    ...
    (yes || x === :end || (HAS_BEGIN_INDEXING && x === :begin) || x === :_)
```
and 
```julia
...
function lower_index 
    ...
    elseif (HAS_BEGIN_INDEXING && index === :begin)
            if dim === nothing
                return :($(Base.firstindex)($collection))
            else
                return :($(Base.firstindex)($collection, $dim))
            end
```

Reply:
That's a good idea; will do that:)